src/attributes.jl

```diff
@@ -879,6 +901,7 @@ struct ListOfConstraintIndices{F,S} <: AbstractModelAttribute end
 A model attribute for the number of constraints of the type `F`-in-`S` present in the model.
 """
 struct NumberOfConstraints{F,S} <: AbstractModelAttribute end
+attribute_value_type(::NumberOfConstraints) = Int
```

Comment:
Why do we use `Int64` in the indices if the type of `NumberOfConstraints` is `Int`?

Reply:
I think the argument was just for consistency. I can make `NumberOfConstraints` and `NumberOfVariables` `Int64` if we want. 

I don't know if anyone has actually solved a model with more than `typemax(Int32)` variables or constraints in JuMP, because CPLEX and Gurobi only wrap the 32-bit API of the solvers, and no one has complained...

Reply:
We could also use `Union{Int64, Int}`, on a 64-bit computer, we have `Union{Int64, Int} == Int64` anyway.
If we don't allow returning a number of variables of `Int64`, it's no use to allow the indices to be `Int64`.

Reply:
Are there any solvers which actually return `Int64`? We could always change it to `Int64` later, if we have evidence that people are solving problems on 32-bit machines with more than `typemax(Int32)` variables...

Reply:
I know that people have complained to FICO Xpress about the Int32 limit on constraints and they added Int64 methods.
Ex. https://www.fico.com/fico-xpress-optimization/docs/latest/solver/optimizer/HTML/XPRSaddcols.html

Reply:
Anyway, we should use the same type than the indices. If we were to use `Int` instead of `Int64` then we should change indices as well